src/Xrefcheck/Verify.hs

```diff
@@ -327,5 +360,49 @@ checkExternalResource VerifyConfig{..} link
         HttpExceptionRequest _ exc -> case exc of
           StatusCodeException resp _
             | isAllowedErrorCode (statusCode $ responseStatus resp) -> Right ()
-            | otherwise -> Left $ ExternalResourceUnavailable (responseStatus resp)
+            | otherwise -> Left $ ExternalHttpResourceUnavailable (responseStatus resp)
           other -> Left . ExternalResourceSomeError $ show other
+
+    checkFtp :: URI -> Bool -> ExceptT VerifyError IO ()
+    checkFtp uri secure = do
+      -- get authority which stores host and port
+      authority <- case uriAuthority uri of
+        Right a -> pure a
+        Left _ -> throwError $
+          ExternalResourceInvalidUrl (Just "FTP path must be absolute")
+      let host = T.unpack . unRText $ authHost authority
+      let port :: Int = fromIntegral . fromMaybe 21 $ authPort authority
+      -- build path from pieces
+      path <- case uriPath uri of
+        Nothing -> pure "/"
+        Just (_, pieces) -> pure . T.unpack . mconcat
+          . NonEmpty.toList
+          . NonEmpty.intersperse "/"
+          . NonEmpty.map unRText
+          $ pieces
+
+      makeFtpRequest host port path secure `catch` \e ->
+        throwError $ ExternalFtpException e
+
+    makeFtpRequest
+      :: String
+      -> Int
+      -> FilePath
+      -> Bool
+      -> ExceptT VerifyError IO ()
+    makeFtpRequest host port path secure = handler host port $
+      \handle response -> do
+        -- check connection status
+        when (frStatus response /= Success) $
+          throwError $ ExternalFtpResourceUnavailable response
+        -- anonymous login
+        loginResp <- login handle "anonymous" ""
+        -- check login status
+        when (frStatus loginResp /= Success) $
+          throwError $ ExternalFtpException $ UnsuccessfulException loginResp
+        file <- nlst handle [ path ]
+        -- check file exists
+        when (BS.null file) $
+          throwError $ FileDoesNotExist path
```

Comment:
And a minor thing: my opinion is that it's worth using a dedicated error for this and have separate `LocalFileDoesNotExist` and `FtpFileDoesNotExist`.

I tended to prefix error message with `⛀` for local files and `⛂` for remote resources, and now we got an exception from this rule which may cause a confusion to the user.

Reply:
Can't see your symbols(
<img width="217" alt="image" src="https://user-images.githubusercontent.com/6579275/127623277-f8f0bf6e-18f1-41b8-befa-8426b4479318.png">

is it a font?

Reply:
Wow that's super weird, I used just plan text there. For me it looks like 
![Screenshot from 2021-07-30 16-56-36](https://user-images.githubusercontent.com/5394217/127663760-d5aba870-7b86-4067-9dd8-08bf228b4a59.png)

Reply:
I can copy it and search, and google says its uff-8 symbol.. but I can't see it (same in source code)
<img width="641" alt="image" src="https://user-images.githubusercontent.com/6579275/127671681-88a03b8b-1cf3-4d0d-9952-2e63292560a2.png">

what OS do you have?

Reply:
<img width="208" alt="image" src="https://user-images.githubusercontent.com/6579275/127671836-0588cb96-a237-478c-96ab-ca75de20ae16.png">

Reply:
Oh sorry, I looked at the wrong position, I see what you mean now.

That's what I get:

![Screenshot from 2021-07-30 17-59-11](https://user-images.githubusercontent.com/5394217/127672308-32ea3a49-7f15-4299-b5b1-f5b7654299c6.png)

I'm on Ubuntu 18, there these symbols render well for me in IDEs and in terminal,

Reply:
can you check what is this font here?
https://fedoraproject.org/wiki/Identifying_fonts

anyway
is it good idea to use such a symbols in our code? maybe someone have same problem

Reply:
Here is this symbol, looks like fonts that support it are also mentioned there: https://www.fileformat.info/info/unicode/char/26c2/index.htm

> is it good idea to use such a symbols in our code? maybe someone have same problem

It is my little wish to use such fancy things in this project :stuck_out_tongue: 

In the worst case, you can just blindly copy-paste the symbols.

Reply:
Check it please